app/transcription.py

```
@@ -275,10 +282,20 @@ def check_if_youtube(source: Source):
                 )
-            if source.source_file.lower().endswith((".mp4", ".webm", ".mov")):
+            if lower_url.endswith((".mp4", ".webm", ".mov")):
                 # regular remote video, not youtube
                 source.preprocess = False
                 return Video(source=source)
+                
+            # If it's a web page that is not a known YouTube hostname, use Scraper.
+            # Use hostname (not netloc) and an exact allowlist to prevent bypass via
+            # subdomains like "youtube.com.evil.com" or "www.notyoutube.com".
+            _YOUTUBE_HOSTS = {"youtube.com", "www.youtube.com", "m.youtube.com", "youtu.be"}
+            parsed_url = urllib.parse.urlparse(lower_url)
```

Ruff (0.15.17)

[error] Undefined name `urllib`

(F821)